boa3/internal/model/builtin/classmethod/replacemethod.py

```diff
@@ -0,0 +1,179 @@
+import ast
+from typing import Any, Dict, Optional, Iterable
+
+from boa3.internal.model.builtin.method.builtinmethod import IBuiltinMethod
+from boa3.internal.model.type.primitive.ibytestringtype import IByteStringType
+from boa3.internal.model.variable import Variable
+from boa3.internal.neo.vm.opcode.Opcode import Opcode
+
+
+class ReplaceMethod(IBuiltinMethod):
+    def __init__(self, self_type: IByteStringType = None):
+        from boa3.internal.model.type.type import Type
+
+        if not isinstance(self_type, IByteStringType):
+            self_type = Type.bytes
```

Comment:
Since `self_type` is an `IByteStringType` it would be nice to have `bytes` tests too, not just `str` tests.